language/vm/vm_runtime/src/runtime.rs

```diff
@@ -112,14 +113,13 @@ impl<'alloc> VMRuntime<'alloc> {
         txn_block: Vec<Transaction>,
         data_view: &dyn StateView,
     ) -> Vec<TransactionOutput> {
-        let mut txns = vec![];
-        for txn in txn_block {
-            match txn {
-                Transaction::UserTransaction(user_txn) => txns.push(user_txn),
-                // TODO: Refactor the logic for writeset transaction.
-                _ => return Vec::new(),
-            }
-        }
+        let txns = txn_block
+            .into_iter()
+            .map(|txn| {
+                SignedTransaction::try_from(txn)
+                    .expect("All transaction should be user transaction")
+            })
+            .collect();
```

Comment:
let txns = txn_block.into_iter().map(SignedTransaction::try_from).collect::<Result<Vec<_>>>().expect("...");

Reply:
I tried to do that. The problem is the `vm_runtime` crate doesn't use the failure crate. So there's no `Result` type there.

Reply:
Just import the failure crate since you already have the dependency? I don't have strong opinion on this though. I guess we'll change this eventually.

Reply:
My plan is to leave it as it is and get rid of the `expect` soon in the PRs to come.